lib/vagrant-service-manager/services/docker.rb

```diff
@@ -71,6 +69,35 @@ def self.print_env_info(ui, options)
           PluginUtil.print_shell_configure_info(ui, ' docker')
         end
       end
+
+      def self.install_cli(machine, ui, options)
+        Installer.new('docker', machine, ui, options).run
+      end
+
+      def self.binary_path(version)
+        "#{BIN_FOLDER}/docker/#{version}/docker"
+      end
+
+      def self.binary_download_url(machine, version)
+        arch = machine.env.host.capability(:os_arch)
+        "#{BINARY_BASE_URL}/#{os_type}/#{arch}/docker-#{version}"
```

Comment:
Is this not exactly what we wanted to avoid? Now this service class knows how to download for ADB. What's about the CDK? Was the idea not to encapsulate this?